cpp/simulations/munich_graph_sim.cpp

```diff
@@ -0,0 +1,297 @@
+                                   mio::osecir::InfectionState::InfectedNoSymptoms,
+                                   mio::osecir::InfectionState::InfectedSymptoms,
+                                   mio::osecir::InfectionState::Recovered};
+    BOOST_OUTCOME_TRY(mobility_data,
+                      mio::read_mobility_plain(mio::path_join((data_dir).string(), "mobility_matrix.txt")));
+    if (mobility_data.rows() != Eigen::Index(params_graph.nodes().size()) ||
+        mobility_data.cols() != Eigen::Index(params_graph.nodes().size())) {
+        return mio::failure(mio::StatusCode::InvalidValue,
+                            "Mobility matrices do not have the correct size. You may need to run "
+                            "transformMobilitydata.py from pycode memilio epidata package.");
+    }
+
+    for (size_t node_i = 0; node_i < params_graph.nodes().size(); ++node_i) {
+        for (size_t node_j = 0; node_j < params_graph.nodes().size(); ++node_j) {
+            auto& populations    = params_graph.nodes()[node_i].property.populations;
+            auto mobility_coeffs = mio::MigrationCoefficientGroup(1, populations.numel());
+
+            auto coeff =
+                mobility_data(node_i, node_j) < 1 ? 0 : mobility_data(node_i, node_j) / populations.get_total();
+
+            for (auto age = mio::AgeGroup(0); age < populations.template size<mio::AgeGroup>(); ++age) {
+                for (auto compartment : migrating_compartments) {
+                    auto coeff_idx                               = populations.get_flat_index({age, compartment});
+                    mobility_coeffs[0].get_baseline()[coeff_idx] = coeff;
+                }
+                if (coeff > 1e-5) {
+                    params_graph.add_edge(node_i, node_j, std::move(mobility_coeffs));
+                }
+            }
+        }
+    }
+    return mio::success();
+}
+
+/**
+ *
+*/
+mio::IOResult<mio::Graph<mio::osecir::Model, mio::MigrationParameters>> get_graph(mio::Date start_date,
+                                                                                  const fs::path& data_dir)
+{
+    const int num_age_groups = 1;
+    mio::osecir::Parameters params(num_age_groups);
+    params.get<mio::osecir::StartDay>() = mio::get_day_in_year(start_date);
+
+    //set covid parameters
+    BOOST_OUTCOME_TRY(set_covid_parameters(params));
+    //set contact matrices
+    double contact_freq                     = 10;
+    mio::ContactMatrixGroup& contact_matrix = params.get<mio::osecir::ContactPatterns>();
+    contact_matrix[0]                       = mio::ContactMatrix(Eigen::MatrixXd::Constant(1, 1, contact_freq));
+    //set npis
+    contact_matrix.add_damping(0.7, mio::SimulationTime(30));
+    contact_matrix.add_damping(0.1, mio::SimulationTime(50));
+
+    mio::Graph<mio::osecir::Model, mio::MigrationParameters> params_graph;
+    //set nodes
```

Comment:
I think all these short comments are quite self-explanatory and could be omitted.